src/sentry/eventstream/base.py

```diff
@@ -71,20 +78,24 @@ def _dispatch_post_process_group_task(
         primary_hash: Optional[str],
         skip_consume: bool = False,
         group_states: Optional[GroupStates] = None,
+        message_type: str = "error",
```

Comment:
I don't think message_type "error" is a reasonable default here, please ensure a value is passed instead.

Reply:
> Nearly all processing uses `Event.get_event_type() == "transaction"` to differentiate between errors and transactions case. While it may not be ideal, I think it's better to stick to this than try to pass another message type parameter all the way through a bunch of functions for the sole purpose of selecting a queue.
> 
> As for separate tasks, I think this is likely the long term direction. The eventstream only supports errors and transaction types and it would be a pretty herculean effort to get another event type in there and I don't think we would do that. All of our new event types never piggyback the eventstream abstraction anymore. More likely it will actually split into separate errors and transactions variants anyway (perhaps something along the lines of #37376) and thus a separate task may be desirable. But I won't block on this if you want to keep it the same right now.

I don't mind what we pass too much, but we have to pass something through the headers to the ppf so that it can differentiate and decide which queue to pass to. I'm happy for that to just be the event type. 

For separate tasks, we might not be using the ppf here but we are likely sticking with the same task that will be fired from somewhere else. We'll see how this evolves as we work with the profiling team though.